internal/database/transactions.go

```diff
@@ -0,0 +1,43 @@
+package database
+
+import (
+	"database/sql"
+	"errors"
+
+	"go.uber.org/zap"
+)
+
+// Transaction represents an in flight change being made to the database that must be committed or rolled back.
+type Transaction[T any] struct {
+	logger *zap.SugaredLogger
+	tx     *sql.Tx
+
+	Record T
+}
```

Comment:
A transaction on a single record seems strange to me. What if I want a transaction on many things?